From: Jan Kara <jack@suse.com>
Subject: xfs: Fix softlockup in xfs_inode_ag_walk()
References: bsc#948347
Patch-mainline: Never (part of 8daaa83145ef1f0a146680618328dbbd0fa76939)

xfs_inode_ag_walk() can scan significant amounts of inodes without scheduling
which leads to softlockup reports. Fix the problem by adding cond_resched()
into appropriate place.

Signed-off-by: Jan Kara <jack@suse.com>

---
 fs/xfs/linux-2.6/xfs_sync.c |    1 +
 1 file changed, 1 insertion(+)

--- a/fs/xfs/linux-2.6/xfs_sync.c
+++ b/fs/xfs/linux-2.6/xfs_sync.c
@@ -191,6 +191,7 @@ restart:
 		if (error == EFSCORRUPTED)
 			break;
 
+		cond_resched();
 	} while (nr_found && !done);
 
 	if (skipped) {
